testing/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.20.7
+FROM golang:1.21
```

Comment:
should this keep with `major.minor.patch` convention, `1.21.0`?

Reply:
yeah, we can do that. good call.